src/Silverback.Integration/Messaging/Encryption/SymmetricDecryptStream.cs

```diff
@@ -65,38 +65,32 @@ protected override void Dispose(bool disposing)
             Check.NotNull(settings, nameof(settings));
             Check.NotNull(stream, nameof(stream));
 
-            byte[]? encryptionKey;
+            byte[]? encryptionKey = settings.KeyProvider != null
+                ? settings.KeyProvider(keyIdentifier ?? string.Empty)
```

Comment:
Why not forward `null`?

Reply:
Uhm yes there is no special reason to forward string.Empty. I thought it was better but I can change it if you prefer 😃